Api/Data/RecordInterface.php

```diff
@@ -17,9 +17,9 @@ interface RecordInterface
 
     /**
      * @param string $channel
-     * @return $this
+     * @return RecordInterface
      */
-    public function setChannel(string $channel);
+    public function setChannel(string $channel): RecordInterface;
```

Comment:
`self` as a return type works, since 7 or so... Buuut `self` doesn't enforce `return $this` or anything, so I don't think it matters too much in terms of PHP: https://3v4l.org/MMBZq

That said, I think `: self` / `@return $this` communicates unambiguously to programmers that the setter returns itself. I'll leave it up to you.

Reply:
Breaking news! The `self` return type gets _real weird_ once inheritance is involved. https://3v4l.org/ldiXd Probably what you've got is the most reliable.